{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open('plus ssRNA.txt', 'r')\n",
    "x = f.readlines()\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "x2 = []\n",
    "\n",
    "for i in x:\n",
    "    if 'virus' in i and 'viridae' not in i:\n",
    "        x2.append(i.split('-')[0][:-1])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "900"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(x2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open('minus ssRNA.txt', 'r')\n",
    "x = f.readlines()\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "x3 = []\n",
    "\n",
    "for i in x:\n",
    "    if 'virus' in i and 'viridae' not in i:\n",
    "        x3.append(i.split('-')[0][:-1])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ViRAD (-)\\n',\n",
       " 'Virus (re-)annotation database\\n',\n",
       " \"Note: The alignment 'depth' is encoded in the link: (M) - Genome map only (no alignment produced). (?.?) - Synonymous site conservation and coding potential plots produced; the number '?.?' gives an indication of the alignment 'depth' and therefore statistical power (essentially it is the mean z-score that would be obtained for 25-codon window with 100% suppression of synonymous substitutions, prior to correction for multiple testing). Alignments with scores < 3 are virtually useless for the synplot2 analysis but some of these may still provide useful MLOGD results.\\n\",\n",
       " '\\n',\n",
       " '\\n',\n",
       " 'Arenaviridae -\\n',\n",
       " '\\n',\n",
       " 'Boa arenavirus NL - Seg-L (4.5), Seg-S (4.0)\\n',\n",
       " 'CAS virus - Seg-L (M), Seg-S (M)\\n',\n",
       " 'Golden Gate virus - Seg-L (4.2), Seg-S (M)\\n',\n",
       " 'University of Helsinki virus - Seg-L (2.7), Seg-S (M)\\n',\n",
       " '\\n',\n",
       " 'Arenaviridae - Arenavirus\\n',\n",
       " '\\n',\n",
       " 'Lujo virus - Seg-L (M), Seg-S (M)\\n',\n",
       " 'Luna virus - Seg-L (5.6), Seg-S (12.9)\\n',\n",
       " '\\n',\n",
       " 'Arenaviridae - Arenavirus New world arenaviruses\\n',\n",
       " '\\n',\n",
       " 'Allpahuayo virus - Seg-L (M), Seg-S (8.7)\\n',\n",
       " 'Amapari virus - Seg-L (0.2), Seg-S (7.3)\\n',\n",
       " 'Bear Canyon virus - Seg-L (2.5), Seg-S (9.6)\\n',\n",
       " 'Chapare virus - Seg-L (M), Seg-S (4.6)\\n',\n",
       " 'Cupixi virus - Seg-L (M), Seg-S (M)\\n',\n",
       " 'Flexal virus - Seg-L (M), Seg-S (8.7)\\n',\n",
       " 'Guanarito virus - Seg-L (0.7), Seg-S (8.2)\\n',\n",
       " 'Junin virus - Seg-L (2.3), Seg-S (7.4)\\n',\n",
       " 'Latino virus - Seg-L (M), Seg-S (4.7)\\n',\n",
       " 'Machupo virus - Seg-L (3.2), Seg-S (8.2)\\n',\n",
       " 'Oliveros virus - Seg-L (M), Seg-S (4.7)\\n',\n",
       " 'Parana virus - Seg-L (M), Seg-S (8.3)\\n',\n",
       " 'Pichinde virus - Seg-L (3.5), Seg-S (8.6)\\n',\n",
       " 'Pirital virus - Seg-L (1.8), Seg-S (9.2)\\n',\n",
       " 'Sabia virus - Seg-L (M), Seg-S (4.6)\\n',\n",
       " 'Tacaribe virus - Seg-L (M), Seg-S (6.7)\\n',\n",
       " 'Tamiami virus - Seg-L (2.4), Seg-S (4.2)\\n',\n",
       " 'Whitewater Arroyo virus - Seg-L (4.4), Seg-S (10.2)\\n',\n",
       " '\\n',\n",
       " 'Arenaviridae - Arenavirus Old world arenaviruses\\n',\n",
       " '\\n',\n",
       " 'Ippy virus - Seg-L (M), Seg-S (M)\\n',\n",
       " 'Lassa virus - Seg-L (8.5), Seg-S (14.1)\\n',\n",
       " 'Lunk virus NKS-1 - Seg-L (M), Seg-S (14.6)\\n',\n",
       " 'Lymphocytic choriomeningitis virus - Seg-L (12.4), Seg-S (14.8)\\n',\n",
       " 'Merino Walk virus - Seg-L (M), Seg-S (M)\\n',\n",
       " 'Mobala virus - Seg-L (M), Seg-S (12.3)\\n',\n",
       " 'Mopeia Lassa virus reassortant 29 - Seg-L (M), Seg-S (14.1)\\n',\n",
       " 'Mopeia virus AN20410 - Seg-L (M), Seg-S (13.9)\\n',\n",
       " 'Morogoro virus - Seg-L (M), Seg-S (12.8)\\n',\n",
       " '\\n',\n",
       " 'Bunyaviridae -\\n',\n",
       " '\\n',\n",
       " 'Lone Star virus - Seg-L (M), Seg-M (M), Seg-S (M)\\n',\n",
       " '\\n',\n",
       " 'Bunyaviridae - Hantavirus\\n',\n",
       " '\\n',\n",
       " 'Andes virus - Seg-L (17.7), Seg-M (16.1), Seg-S (24.2)\\n',\n",
       " 'Dobrava-Belgrade virus - Seg-L (14.2), Seg-M (18.1), Seg-S (17.6)\\n',\n",
       " 'Hantaan virus - Seg-L (14.2), Seg-M (18.1), Seg-S (17.7)\\n',\n",
       " 'Hantavirus Z10 - Seg-L (14.2), Seg-M (18.1), Seg-S (17.6)\\n',\n",
       " 'Puumala virus - Seg-L (17.9), Seg-M (14.0), Seg-S (21.9)\\n',\n",
       " 'Seoul virus - Seg-L (14.2), Seg-M (18.0), Seg-S (18.5)\\n',\n",
       " 'Sin Nombre virus - Seg-L (17.9), Seg-M (17.5), Seg-S (21.4)\\n',\n",
       " 'Thottapalayam virus - Seg-L (5.1), Seg-M (4.1), Seg-S (4.4)\\n',\n",
       " 'Tula virus - Seg-L (17.9), Seg-M (14.0), Seg-S (24.2)\\n',\n",
       " '\\n',\n",
       " 'Bunyaviridae - Nairovirus\\n',\n",
       " '\\n',\n",
       " 'Crimean-Congo hemorrhagic fever virus - Seg-L (9.1), Seg-M (10.0), Seg-S (10.3)\\n',\n",
       " 'Dugbe virus - Seg-L (4.6), Seg-M (M), Seg-S (5.0)\\n',\n",
       " '\\n',\n",
       " 'Bunyaviridae - Orthobunyavirus\\n',\n",
       " '\\n',\n",
       " 'Aino virus - Seg-L (5.9), Seg-M (5.6), Seg-S (9.6)\\n',\n",
       " 'Akabane virus - Seg-L (5.1), Seg-M (5.8), Seg-S (9.7)\\n',\n",
       " 'Brazoran virus - Seg-L (M), Seg-M (M), Seg-S (M)\\n',\n",
       " 'Bunyamwera virus - Seg-L (8.8), Seg-M (4.6), Seg-S (10.9)\\n',\n",
       " 'Cat Que virus - Seg-L (7.5), Seg-M (6.5), Seg-S (10.8)\\n',\n",
       " 'La Crosse virus - Seg-L (11.2), Seg-M (11.9), Seg-S (11.9)\\n',\n",
       " 'Murrumbidgee virus - Seg-L (M), Seg-M (M), Seg-S (M)\\n',\n",
       " 'Oropouche virus - Seg-L (5.8), Seg-M (1.2), Seg-S (11.9)\\n',\n",
       " 'Sathuperi virus - Seg-L (4.9), Seg-M (5.1), Seg-S (9.8)\\n',\n",
       " 'Shamonda virus - Seg-L (4.9), Seg-M (2.9), Seg-S (9.9)\\n',\n",
       " 'Simbu virus - Seg-L (M), Seg-M (M), Seg-S (9.6)\\n',\n",
       " '\\n',\n",
       " 'Bunyaviridae - Phlebovirus\\n',\n",
       " '\\n',\n",
       " 'Aguacate virus - Seg-L (6.1), Seg-M (M), Seg-S (6.1)\\n',\n",
       " 'Arumowot virus - Seg-L (4.7), Seg-M (M), Seg-S (M)\\n',\n",
       " 'Candiru virus - Seg-L (9.8), Seg-M (M), Seg-S (9.5)\\n',\n",
       " 'Razdan virus - Seg-L (6.7), Seg-M (7.0), Seg-S (6.3)\\n',\n",
       " 'Rift Valley fever virus - Seg-L (4.7), Seg-M (5.0), Seg-S (6.7)\\n',\n",
       " 'Sandfly fever Naples virus - Seg-L (9.3), Seg-M (4.6), Seg-S (7.7)\\n',\n",
       " 'Sandfly fever Turkey virus - Seg-L (M), Seg-M (2.4), Seg-S (5.0)\\n',\n",
       " 'SFTS virus HB29 - Seg-L (5.9), Seg-M (5.6), Seg-S (8.7)\\n',\n",
       " 'Uukuniemi virus - Seg-L (9.1), Seg-M (7.1), Seg-S (M)\\n',\n",
       " '\\n',\n",
       " 'Bunyaviridae - Tospovirus\\n',\n",
       " '\\n',\n",
       " 'Bean necrotic mosaic virus - Seg-L (M), Seg-M (M), Seg-S (M)\\n',\n",
       " 'Capsicum chlorosis virus - Seg-L (8.8), Seg-M (7.5), Seg-S (10.3)\\n',\n",
       " 'Groundnut bud necrosis virus - Seg-L (8.8), Seg-M (7.4), Seg-S (9.9)\\n',\n",
       " 'Groundnut ringspot and Tomato chlorotic spot virus reassortant - Seg-L (6.0), Seg-M (7.3), Seg-S (7.8)\\n',\n",
       " 'Impatiens necrotic spot virus - Seg-L (1.1), Seg-M (1.4), Seg-S (1.7)\\n',\n",
       " 'Melon yellow spot virus - Seg-L (7.7), Seg-M (1.7), Seg-S (2.7)\\n',\n",
       " 'Tomato spotted wilt virus - Seg-L (5.9), Seg-M (7.3) (replacement RefSeq), Seg-M (M), Seg-S (7.7)\\n',\n",
       " 'Tomato zonate spot virus - Seg-L (8.2), Seg-M (5.6), Seg-S (3.7)\\n',\n",
       " 'Watermelon silver mottle virus - Seg-L (8.8), Seg-M (7.3), Seg-S (9.7)\\n',\n",
       " '\\n',\n",
       " 'Emaravirus\\n',\n",
       " '\\n',\n",
       " 'European mountain ash ringspot-associated virus - RNA1 (M), RNA2 (M), RNA3 (6.1), RNA4 (M)\\n',\n",
       " 'Rose rosette virus - RNA1 (M), RNA2 (M), RNA3 (2.4), RNA4 (2.5)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales -\\n',\n",
       " '\\n',\n",
       " 'Midway virus - (M)\\n',\n",
       " 'Nyamanini virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Bornaviridae - Bornavirus\\n',\n",
       " '\\n',\n",
       " 'Borna disease virus - (4.7)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Filoviridae -\\n',\n",
       " '\\n',\n",
       " 'Lloviu virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Filoviridae - Ebolavirus\\n',\n",
       " '\\n',\n",
       " 'Bundibugyo ebolavirus - (5.8)\\n',\n",
       " 'Ebola virus - Mayinga Zaire 1976 - (5.8)\\n',\n",
       " 'Reston ebolavirus - (1.7)\\n',\n",
       " 'Sudan ebolavirus - (1.9)\\n',\n",
       " 'Tai Forest ebolavirus - (5.8)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Filoviridae - Marburgvirus\\n',\n",
       " '\\n',\n",
       " 'Marburg marburgvirus - (4.7)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae -\\n',\n",
       " '\\n',\n",
       " 'Nariva virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae -\\n',\n",
       " '\\n',\n",
       " 'Beilong virus - (4.6)\\n',\n",
       " 'J-virus - (M)\\n',\n",
       " 'Mossman virus - (M)\\n',\n",
       " 'Tupaia paramyxovirus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae - Avulavirus\\n',\n",
       " '\\n',\n",
       " 'Avian paramyxovirus 4 - (4.1)\\n',\n",
       " 'Avian paramyxovirus 6 - (5.2)\\n',\n",
       " 'Goose paramyxovirus SF02 - (13.9)\\n',\n",
       " 'Newcastle disease virus B1 - (13.8)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae - Ferlavirus\\n',\n",
       " '\\n',\n",
       " 'Fer-de-Lance paramyxovirus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae - Henipavirus\\n',\n",
       " '\\n',\n",
       " 'Hendra virus - (5.0)\\n',\n",
       " 'Nipah virus - (5.0)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae - Morbillivirus\\n',\n",
       " '\\n',\n",
       " 'Canine distemper virus - (5.4)\\n',\n",
       " 'Dolphin morbillivirus - (M)\\n',\n",
       " 'Measles virus - (6.1)\\n',\n",
       " 'Peste-des-petits-ruminants virus - (3.2)\\n',\n",
       " 'Rinderpest virus strain Kabete O - (5.5)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae - Respirovirus\\n',\n",
       " '\\n',\n",
       " 'Bovine parainfluenza virus 3 - (7.2)\\n',\n",
       " 'Human parainfluenza virus 1 - (6.1)\\n',\n",
       " 'Human parainfluenza virus 3 - (7.1)\\n',\n",
       " 'Sendai virus - (6.1)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Paramyxovirinae - Rubulavirus\\n',\n",
       " '\\n',\n",
       " 'Human parainfluenza virus 2 - (1.8)\\n',\n",
       " 'Human parainfluenza virus 4a - (3.2)\\n',\n",
       " 'Mapuera virus - (M)\\n',\n",
       " 'Menangle virus - (1.8)\\n',\n",
       " 'Mumps virus - (6.0)\\n',\n",
       " 'Parainfluenza virus 5 - (1.9)\\n',\n",
       " 'Porcine rubulavirus - (M)\\n',\n",
       " 'Simian virus 41 - (M)\\n',\n",
       " 'Tioman virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Pneumovirinae - Metapneumovirus\\n',\n",
       " '\\n',\n",
       " 'Avian metapneumovirus - (2.2)\\n',\n",
       " 'Human metapneumovirus - (7.1)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Paramyxoviridae - Pneumovirinae - Pneumovirus\\n',\n",
       " '\\n',\n",
       " 'Bovine respiratory syncytial virus - (7.0)\\n',\n",
       " 'Human respiratory syncytial virus - (7.0)\\n',\n",
       " 'Pneumonia virus of mice J3666 - (1.5)\\n',\n",
       " 'Respiratory syncytial virus - (6.9)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae -\\n',\n",
       " '\\n',\n",
       " 'Ngaingan virus - (M)\\n',\n",
       " 'Siniperca chuatsi rhabdovirus - (1.6)\\n',\n",
       " 'Tupaia virus - (M)\\n',\n",
       " 'Wongabel virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Cytorhabdovirus\\n',\n",
       " '\\n',\n",
       " 'Lettuce necrotic yellows virus - (M)\\n',\n",
       " 'Lettuce yellow mottle virus - (M)\\n',\n",
       " 'Northern cereal mosaic virus - (2.2)\\n',\n",
       " 'Persimmon virus A - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Ephemerovirus\\n',\n",
       " '\\n',\n",
       " 'Bovine ephemeral fever virus - (M)\\n',\n",
       " 'Kotonkan virus - (M)\\n',\n",
       " 'Obodhiang virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Lyssavirus\\n',\n",
       " '\\n',\n",
       " 'Aravan virus - (19.5)\\n',\n",
       " 'Australian bat lyssavirus - (18.9)\\n',\n",
       " 'Duvenhage virus - (18.7)\\n',\n",
       " 'European bat lyssavirus 1 - (19.5)\\n',\n",
       " 'European bat lyssavirus 2 - (19.0)\\n',\n",
       " 'Ikoma lyssavirus - (M)\\n',\n",
       " 'Irkut virus - (19.6)\\n',\n",
       " 'Lagos bat virus - (13.5)\\n',\n",
       " 'Mokola virus - (10.1)\\n',\n",
       " 'Rabies virus - (18.9)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Novirhabdovirus\\n',\n",
       " '\\n',\n",
       " 'Hirame rhabdovirus - (4.7)\\n',\n",
       " 'Infectious hematopoietic necrosis virus - (4.7)\\n',\n",
       " 'Snakehead virus - (M)\\n',\n",
       " 'Viral hemorrhagic septicemia virus Fil3 - (4.3)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Nucleorhabdovirus\\n',\n",
       " '\\n',\n",
       " 'Maize fine streak virus - (M)\\n',\n",
       " 'Maize Iranian mosaic virus - (M)\\n',\n",
       " 'Maize mosaic virus - (M)\\n',\n",
       " 'Potato yellow dwarf virus - (M)\\n',\n",
       " 'Rice yellow stunt virus - (0.8)\\n',\n",
       " 'Sonchus yellow net virus - (0.1)\\n',\n",
       " 'Taro vein chlorosis virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Perhabdovirus\\n',\n",
       " '\\n',\n",
       " 'Eel virus European X - (1.1)\\n',\n",
       " 'Perch rhabdovirus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Sigmavirus\\n',\n",
       " '\\n',\n",
       " 'Drosophila melanogaster sigmavirus AP30 - (2.5)\\n',\n",
       " 'Drosophila obscura sigmavirus 10A - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Tibrovirus\\n',\n",
       " '\\n',\n",
       " 'Tibrogargan virus - (M)\\n',\n",
       " '\\n',\n",
       " 'Mononegavirales - Rhabdoviridae - Vesiculovirus\\n',\n",
       " '\\n',\n",
       " 'American bat vesiculovirus TFFN-2013 - (M)\\n',\n",
       " 'Chandipura virus - (6.0)\\n',\n",
       " 'Isfahan virus - (M)\\n',\n",
       " 'Spring viraemia of carp virus - (6.7)\\n',\n",
       " 'Vesicular stomatitis Indiana virus - (5.1)\\n',\n",
       " '\\n',\n",
       " 'Ophioviridae - Ophiovirus\\n',\n",
       " '\\n',\n",
       " 'Citrus psorosis virus - RNA1 (4.1), RNA2 (4.2), RNA3 (3.7)\\n',\n",
       " 'Lettuce ring necrosis virus - RNA1 (M), RNA2 (M), RNA3 (M), RNA4 (M)\\n',\n",
       " 'Mirafiori lettuce big-vein virus - RNA1 (M), RNA2 (1.4), RNA3 (4.3), RNA4 (M)\\n',\n",
       " '\\n',\n",
       " 'Orthomyxoviridae - Influenzavirus A\\n',\n",
       " '\\n',\n",
       " 'Influenza A virus A-Puerto Rico-8-1934 H1N1 - Seg-1 (18.7), Seg-2 (18.1), Seg-3 (14.8), Seg-4 (10.0), Seg-5 (14.8), Seg-6 (11.1), Seg-7 (12.6), Seg-8 (12.4)\\n',\n",
       " '\\n',\n",
       " 'Orthomyxoviridae - Influenzavirus B\\n',\n",
       " '\\n',\n",
       " 'Influenza B virus - RNA1 (6.9), RNA2 (7.1), RNA3 (6.3), RNA4 (6.6), RNA5 (6.8), RNA6 (5.7), RNA7 (6.2), RNA8 (5.2)\\n',\n",
       " '\\n',\n",
       " 'Orthomyxoviridae - Influenzavirus C\\n',\n",
       " '\\n',\n",
       " 'Influenza C virus C-Ann Arbor-1-50 - Seg-1 (1.7), Seg-2 (1.7), Seg-3 (1.6), Seg-4 (5.4), Seg-5 (2.9), Seg-6 (4.3), Seg-7 (3.5)\\n',\n",
       " '\\n',\n",
       " 'Orthomyxoviridae - Isavirus\\n',\n",
       " '\\n',\n",
       " 'Infectious salmon anemia virus - Seg-1 (5.0), Seg-2 (4.9), Seg-3 (5.5), Seg-4 (7.1), Seg-5 (6.1), Seg-6 (5.2), Seg-7 (3.0) (replacement RefSeq), Seg-7 (3.0), Seg-8 (1.7)\\n',\n",
       " '\\n',\n",
       " 'Orthomyxoviridae - Thogotovirus\\n',\n",
       " '\\n',\n",
       " 'Thogoto virus - Seg-1 (M), Seg-2 (M), Seg-3 (M), Seg-4 (M), Seg-5 (M), Seg-6 (4.9)\\n',\n",
       " '\\n',\n",
       " 'Tenuivirus\\n',\n",
       " '\\n',\n",
       " 'Rice grassy stunt virus - RNA1 (0.5), RNA2 (1.4), RNA3 (2.1), RNA4 (1.3), RNA5 (2.4), RNA6 (1.8)\\n',\n",
       " 'Rice stripe virus - RNA1 (5.1), RNA2 (5.4), RNA3 (8.6), RNA4 (8.3)\\n',\n",
       " '\\n',\n",
       " 'Varicosavirus\\n',\n",
       " '\\n',\n",
       " 'Lettuce big-vein associated virus - RNA1 (1.9), RNA2 (1.8)\\n',\n",
       " '\\n',\n",
       " 'Unclassified viruses\\n',\n",
       " '\\n',\n",
       " 'Orchid fleck virus - RNA1 (2.9), RNA2 (3.2)\\n',\n",
       " '\\n']"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x3[-5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df2 = pd.read_csv('ICTV 2018b.v2.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sort</th>\n",
       "      <th>Realm</th>\n",
       "      <th>Subrealm</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Subkingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Subphylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Subclass</th>\n",
       "      <th>Order</th>\n",
       "      <th>...</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Subgenus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Type Species?</th>\n",
       "      <th>Genome Composition</th>\n",
       "      <th>Last Change</th>\n",
       "      <th>MSL of Last Change</th>\n",
       "      <th>Proposal for Last Change</th>\n",
       "      <th>Taxon History URL</th>\n",
       "      <th>Unnamed: 22</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Haploviricotina</td>\n",
       "      <td>Chunqiuviricetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Muvirales</td>\n",
       "      <td>...</td>\n",
       "      <td>Yingvirus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Beihai yingvirus</td>\n",
       "      <td>0</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>Moved,</td>\n",
       "      <td>34</td>\n",
       "      <td>2017.006G.A.v3.Riboviria.zip</td>\n",
       "      <td>ICTVonline=201856030</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Haploviricotina</td>\n",
       "      <td>Chunqiuviricetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Muvirales</td>\n",
       "      <td>...</td>\n",
       "      <td>Yingvirus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Charybdis yingvirus</td>\n",
       "      <td>0</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>Moved,</td>\n",
       "      <td>34</td>\n",
       "      <td>2017.006G.A.v3.Riboviria.zip</td>\n",
       "      <td>ICTVonline=201856031</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Haploviricotina</td>\n",
       "      <td>Chunqiuviricetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Muvirales</td>\n",
       "      <td>...</td>\n",
       "      <td>Yingvirus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hubei yingvirus</td>\n",
       "      <td>0</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>Moved,</td>\n",
       "      <td>34</td>\n",
       "      <td>2017.006G.A.v3.Riboviria.zip</td>\n",
       "      <td>ICTVonline=201856032</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Haploviricotina</td>\n",
       "      <td>Chunqiuviricetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Muvirales</td>\n",
       "      <td>...</td>\n",
       "      <td>Yingvirus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sanxia yingvirus</td>\n",
       "      <td>0</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>Moved,</td>\n",
       "      <td>34</td>\n",
       "      <td>2017.006G.A.v3.Riboviria.zip</td>\n",
       "      <td>ICTVonline=201856033</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>Riboviria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Haploviricotina</td>\n",
       "      <td>Chunqiuviricetes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Muvirales</td>\n",
       "      <td>...</td>\n",
       "      <td>Yingvirus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Shahe yingvirus</td>\n",
       "      <td>0</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>Moved,</td>\n",
       "      <td>34</td>\n",
       "      <td>2017.006G.A.v3.Riboviria.zip</td>\n",
       "      <td>ICTVonline=201856034</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Sort      Realm  Subrealm  Kingdom  Subkingdom           Phylum  \\\n",
       "0     1  Riboviria       NaN      NaN         NaN  Negarnaviricota   \n",
       "1     2  Riboviria       NaN      NaN         NaN  Negarnaviricota   \n",
       "2     3  Riboviria       NaN      NaN         NaN  Negarnaviricota   \n",
       "3     4  Riboviria       NaN      NaN         NaN  Negarnaviricota   \n",
       "4     5  Riboviria       NaN      NaN         NaN  Negarnaviricota   \n",
       "\n",
       "         Subphylum             Class  Subclass      Order  ...      Genus  \\\n",
       "0  Haploviricotina  Chunqiuviricetes       NaN  Muvirales  ...  Yingvirus   \n",
       "1  Haploviricotina  Chunqiuviricetes       NaN  Muvirales  ...  Yingvirus   \n",
       "2  Haploviricotina  Chunqiuviricetes       NaN  Muvirales  ...  Yingvirus   \n",
       "3  Haploviricotina  Chunqiuviricetes       NaN  Muvirales  ...  Yingvirus   \n",
       "4  Haploviricotina  Chunqiuviricetes       NaN  Muvirales  ...  Yingvirus   \n",
       "\n",
       "  Subgenus              Species Type Species? Genome Composition Last Change  \\\n",
       "0      NaN     Beihai yingvirus             0           ssRNA(-)      Moved,   \n",
       "1      NaN  Charybdis yingvirus             0           ssRNA(-)      Moved,   \n",
       "2      NaN      Hubei yingvirus             0           ssRNA(-)      Moved,   \n",
       "3      NaN     Sanxia yingvirus             0           ssRNA(-)      Moved,   \n",
       "4      NaN      Shahe yingvirus             0           ssRNA(-)      Moved,   \n",
       "\n",
       "   MSL of Last Change     Proposal for Last Change      Taxon History URL  \\\n",
       "0                  34  2017.006G.A.v3.Riboviria.zip  ICTVonline=201856030   \n",
       "1                  34  2017.006G.A.v3.Riboviria.zip  ICTVonline=201856031   \n",
       "2                  34  2017.006G.A.v3.Riboviria.zip  ICTVonline=201856032   \n",
       "3                  34  2017.006G.A.v3.Riboviria.zip  ICTVonline=201856033   \n",
       "4                  34  2017.006G.A.v3.Riboviria.zip  ICTVonline=201856034   \n",
       "\n",
       "   Unnamed: 22  \n",
       "0          NaN  \n",
       "1          NaN  \n",
       "2          NaN  \n",
       "3          NaN  \n",
       "4          NaN  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
